### PR TITLE
fix(tests): do not send test GET requests with an HTTP request body

### PR DESCRIPTION
**Pull request by floyd-fuh**
It's simply weird to send GET requests that have a Request Content-Type and a body. It also means if there is anything before the WAF that checks conformity of HTTP protocol the test could fail because of that.

Also, there are WAFs simply removing the HTTP body from GET requests, meaning if you run these tests against them, the test would always fail.

### CODE REVIEW

**Review by fzipi — DISMISSED** [review]
I agree this should not be used for rules in the 941XXX range. If we need to have this tested, is probably in the 92XXXX range.

Probably leftovers from copy & paste.

**Review by franbuehler — APPROVED** [review]
Wow! Nice finding! Thanks for providing this fix!

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941140.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ tests:
       - stage:
           input:
             dest_addr: 127.0.0.1
-            method: GET
+            method: POST
             port: 80
             uri: '/foo'
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941140.yaml:16]:
```suggestion
            uri: '/post'
```

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941140.yaml [hunk 2]
```diff
@@ -27,7 +27,7 @@ tests:
       - stage:
           input:
             dest_addr: 127.0.0.1
-            method: GET
+            method: POST
             port: 80
             uri: '/bar'
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941140.yaml:32]:
```suggestion
            uri: '/post'
```

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941150.yaml [hunk 3]
```diff
@@ -11,7 +11,7 @@ tests:
       - stage:
           input:
             dest_addr: 127.0.0.1
-            method: GET
+            method: POST
             port: 80
             uri: '/foo'
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941150.yaml:16]:
```suggestion
            uri: '/post'
```

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941190.yaml [hunk 4]
```diff
@@ -11,7 +11,7 @@ tests:
       - stage:
           input:
             dest_addr: 127.0.0.1
-            method: GET
+            method: POST
             port: 80
             uri: '/foo'
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941190.yaml:16]:
```suggestion
            uri: '/post'
```

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941190.yaml [hunk 5]
```diff
@@ -27,7 +27,7 @@ tests:
       - stage:
           input:
             dest_addr: 127.0.0.1
-            method: GET
+            method: POST
             port: 80
             uri: '/bar'
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941190.yaml:32]:
```suggestion
            uri: '/post'
```

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941200.yaml [hunk 6]
```diff
@@ -11,7 +11,7 @@ tests:
       - stage:
           input:
             dest_addr: 127.0.0.1
-            method: GET
+            method: POST
             port: 80
             uri: '/foo'
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941200.yaml:16]:
```suggestion
            uri: '/post'
```

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941200.yaml [hunk 7]
```diff
@@ -27,7 +27,7 @@ tests:
       - stage:
           input:
             dest_addr: 127.0.0.1
-            method: GET
+            method: POST
             port: 80
             uri: '/bar'
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941200.yaml:32]:
```suggestion
            uri: '/post'
```

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941370.yaml [hunk 8]
```diff
@@ -47,7 +47,7 @@ tests:
               Host: localhost
               User-Agent: "OWASP CRS test agent"
               Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
-            method: GET
+            method: POST
             port: 80
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941370.yaml:51]:
```suggestion
            port: 80
            uri: '/post'
```

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941370.yaml [hunk 9]
```diff
@@ -63,7 +63,7 @@ tests:
...
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941370.yaml:67]:
```suggestion
            port: 80
            uri: '/post'
```

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941370.yaml [hunk 10]
```diff
@@ -79,7 +79,7 @@ tests:
...
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941370.yaml:83]:
```suggestion
            port: 80
            uri: '/post'
```

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941370.yaml [hunk 11]
```diff
@@ -95,7 +95,7 @@ tests:
...
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941370.yaml:99]:
```suggestion
            port: 80
            uri: '/post'
```

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941370.yaml [hunk 12]
```diff
@@ -127,7 +127,7 @@ tests:
...
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941370.yaml:131]:
```suggestion
            port: 80
            uri: '/post'
```

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941370.yaml [hunk 13]
```diff
@@ -143,7 +143,7 @@ tests:
...
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941370.yaml:147]:
```suggestion
            port: 80
            uri: '/post'
```

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941370.yaml [hunk 14]
```diff
@@ -159,7 +159,7 @@ tests:
...
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941370.yaml:163]:
```suggestion
            port: 80
            uri: '/post'
```

**Review by fzipi — COMMENTED** [review]
As @dune73 mentioned, you could also change the uri to `/post`.

**Comment by floyd-fuh** [comment]
sure thing, I changed that

**Review by fzipi — APPROVED** [review]

**Pull request merged**